### PR TITLE
DSK Processing

### PR DESCRIPTION
**Pull request by kris927b**
Adding a generic script for processing DSK data.

The script can so far handle:
- Docx
- PDF
- EPUB
- HTML
- TXT

### CODE REVIEW

File: data-processing/scripts/dsk/convert_dsk_to_jsonl.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    metadata = {**kwargs, **build_metadata(file_path)}
+    return json.dumps(asdict(create_JSONL(text, source, metadata)), ensure_ascii=False)
+
+
+# UNUSED - USING DOCLING INSTEAD
+def process_docx(file_path: Path, source: str, **kwargs: dict[str, Any]) -> str:
+    text = convert_file(file_path, to="plain", format="docx")
+    text = re.sub(r"(\n\s)+", "\n", text)
+    metadata = {**kwargs, **build_metadata(file_path)}
+    return json.dumps(asdict(create_JSONL(text, source, metadata)), ensure_ascii=False)
```
Comment (saattrupdan) [inline data-processing/scripts/dsk/convert_dsk_to_jsonl.py:61]: If these are unused, shouldn't they be removed?

File: data-processing/scripts/dsk/convert_dsk_to_jsonl.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        source (str): What is the data source (most likely DSK client)
+        **kwargs (dict): Additional arguments
+
+    Returns:
+        str: JSONL line with the file content
+    """
+    text = extract_html_text(file_path.read_text())
+    text = re.sub(r"(\n\s)+", "\n", text)
+    metadata = build_metadata(file_path)
+    return json.dumps(asdict(create_JSONL(text, source, metadata)), ensure_ascii=False)
```
Comment (saattrupdan) [inline data-processing/scripts/dsk/convert_dsk_to_jsonl.py:78]: Can't docling handle HTML as well?

File: data-processing/scripts/dsk/utils.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
+from dataclasses import dataclass
```
Comment (saattrupdan) [inline data-processing/scripts/dsk/utils.py:1]: This entire script is really a module, containing functions that you import elsewhere. Shouldn't it belong to `src/dfm_data` for that reason? Then you can import them in your script with `from dfm_data.utils import X`.

The same thing is really the case with all the functions in `convert_dsk_to_jsonl`, except the `crawl_directory` function.

File: data-processing/scripts/dsk/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Data-processing Dansk Sprogmodel Konsortium (DSK)
+
+This directory contains scripts for processing the DSK data. Mainly converting from various formats (e.g. pdf and docx) to jsonl.
+
+Run the conversion on an entire repo:
+
+```bash
+$ python conver_dsk_to_jsonl.py $PATH_TO_DSK_DIR $OUTPUT_PATH $DSK_CLIENT
```
Comment (saattrupdan) [inline data-processing/scripts/dsk/README.md:8]:
```suggestion
$ python convert_dsk_to_jsonl.py $PATH_TO_DSK_DIR $OUTPUT_PATH $DSK_CLIENT
```

File: data-processing/scripts/dsk/convert_dsk_to_jsonl.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        return None
+
+    suffix = file.suffix
+    method = SUFFIX2METHOD.get(suffix)
+
+    if not method:
+        logger.warning(f"Unsupported file type: {suffix} - for file: {file!s}")
+        return None
+
+    return method(file, dsk_client, **kwargs)
```
Comment (saattrupdan) [inline data-processing/scripts/dsk/convert_dsk_to_jsonl.py:192]: The `SUFFIX2METHOD` is only used in `process_file`, so maybe it should be moved in there:

```suggestion
def process_file(file: Path, dsk_client: str, **kwargs: dict) -> str | None:
    if file.is_dir():
        return None

    suffix = file.suffix
    method = {
        ".pdf": process_document,
        ".html": process_html,
        ".docx": process_document,
        ".epub": process_epub,
        ".txt": process_txt,
        ".pptx": process_document,
        ".md": process_document,
    }.get(suffix)

    if not method:
        logger.warning(f"Unsupported file type: {suffix} - for file: {file!s}")
        return None

    return method(file, dsk_client, **kwargs)
```

**Review by saattrupdan — CHANGES_REQUESTED** [review]
A few initial comments.

I can't recall if we've decided on a level of code quality. I would personally like to see docstrings everywhere, including module docstrings, as well as unit tests. But you all think that's a bit too much then disregard that.

File: data-processing/pyproject.toml [hunk 6]
```diff
@@ -22,16 +22,26 @@ dependencies = [
...
   "zstandard",
   "nlp_dedup",
   "datatrove>=0.2.0",
   "pyyaml",
+  "joblib",
+  "pdfminer.six",
+  "pypandoc-binary",
+  "trafilatura",
+  "typer",
+  "loguru",
```
Comment (KennethEnevoldsen) [inline data-processing/pyproject.toml:43]: I would add a lower version on new dependencies

File: data-processing/scripts/dsk/convert_dsk_to_jsonl.py [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+    return json.dumps(asdict(create_JSONL(text, source, metadata)), ensure_ascii=False)
+
+
+def process_html(file_path: Path, source: str, **kwargs: dict[str, Any]) -> str:  # noqa: ARG001
+    """Read a single HTML file and build a JSONL object. Uses Trafilatura for the extraction.
+
+    Args:
+        file_path (Path): Path to the HTML file
+        source (str): What is the data source (most likely DSK client)
+        **kwargs (dict): Additional arguments
```
Comment (KennethEnevoldsen) [inline data-processing/scripts/dsk/convert_dsk_to_jsonl.py:70]: I believe this is a general comment, but let us not duplicate type hints

```suggestion
        file_path: Path to the HTML file
        source: What is the data source (most likely DSK client)
        **kwargs: Additional arguments
```

can lead to errors where the type hints disagree.

File: data-processing/pyproject.toml (hunk 6 above)
Comment (kris927b) [reply data-processing/pyproject.toml]: Added lower bounds in new commit.

File: data-processing/scripts/dsk/utils.py (hunk 3 above)
Comment (kris927b) [reply data-processing/scripts/dsk/utils.py]: Moved all methods except `crawl_directory` to `dfm_data`

**Comment by KennethEnevoldsen** [comment]
Will close this in favor of #302 (feel free to ping me if you don't get a review on PR within a few days)

**Pull request closed without merging**